scripts/import_from_ena_legacy.py

```diff
@@ -0,0 +1,849 @@
+    "WGS": "whole genome sequencing assay",
+    "ChIP-Seq": "ChIP-seq"
+}
+EXPERIMENT_TARGETS = {
+    "ATAC-seq": "open_chromatin_region",
+    "BS-Seq": "DNA methylation",
+    "Hi-C": "chromatin",
+    "DNase": "open_chromatin_region",
+    "RNA-Seq": "Unknown ",
+    "WGS": "input DNA",
+    "ChIP-Seq": "Unknown"
+}
+# value of all is not allowed in the general ena data portal, not like FAANG, so need to list fields
+# which we want to retrieve
+FIELD_LIST = [
+    'study_accession', 'secondary_study_accession', 'sample_accession', 'experiment_accession', 'run_accession',
+    'submission_accession', 'tax_id', 'instrument_platform', 'instrument_model', 'library_strategy',
+    'library_selection', 'read_count', 'base_count', 'first_public', 'last_updated',
+    'study_title', 'study_alias', 'run_alias', 'fastq_bytes', 'fastq_md5',
+    'fastq_ftp', 'fastq_aspera', 'fastq_galaxy', 'submitted_format', 'sra_bytes',
+    'sra_md5', 'sra_ftp', 'sra_aspera', 'sra_galaxy', 'cram_index_ftp',
+    'cram_index_aspera', 'cram_index_galaxy', 'project_name'
+]
+
+# logger = utils.create_logging_instance('import_ena_legacy', level=logging.INFO)
+logger = logging.getLogger(__name__)
+logging.basicConfig(format='%(asctime)s\t%(levelname)s:\t%(name)s line %(lineno)s\t%(message)s', level=logging.INFO)
+
+
+def get_biosamples_records_from_es(host, es_index_prefix, es_type):
+    """
+    Get existing biosample records from elastic search
+    :param host: elasticsearch python instance
+    :param es_index_prefix: the name of the index set
+    :param es_type: type of record, either organism or specimen
+    :return:
+    """
+    global BIOSAMPLES_RECORDS
+    url = f'http://{host}/{es_index_prefix}_{es_type}/_search?size=100000'
+    response = requests.get(url).json()
+    if 'hits' not in response:
+        logger.error(f"No data retrieved from {url}, please double check whether the URL is correct")
+        return
+    for item in response['hits']['hits']:
+        BIOSAMPLES_RECORDS[item['_id']] = item['_source']
+
+
+def get_faang_datasets(host: str, es_index_prefix: str) -> Set[str]:
+    """
+    Get the id list of existing datasets with FAANG standard stored in the Elastic Search
+    :param host: the Elastic Search server address
+    :param es_index_prefix: the Elastic Search dataset index
+    :return: set of dataset id
+    """
+    url = f"http://{host}/{es_index_prefix}_dataset/_search?_source=standardMet"
+    response = requests.get(url).json()
+    total_number = response['hits']['total']
+    if total_number == 0:
+        return set()
+    datasets = set()
+    url = f"{url}&size={total_number}"
+    response = requests.get(url).json()
+    for hit in response['hits']['hits']:
+        if hit['_source']['standardMet'] == 'FAANG':
+            datasets.add(hit['_id'])
+    return datasets
+
+
+def parse_date(date_str: str) -> str:
+    """
+    extract YYYY-MM-DD from ISO date string used by BioSamples
+    :param date_str:
+    :return:
```

Comment:
1. Documentation should be added
2. We already have the same function in import_from_biosamples, so it should be extracted to misc module

Reply:
done in 6981816 for import from biosample

Reply:
complete in 27a573a for import from ena legacy